### PR TITLE
chore:core: add exceeded caps revert tests

### PR DESCRIPTION
**Pull request by nmimran99**
chore:core: add exceeded caps revert tests

Add reverting tests for different scenarios where user/global caps have already exceeded.

### CODE REVIEW

**Review by ramenforbreakfast — APPROVED** [review]
LGTM, besides requested changes

File: apps/smart-contracts/core/test/unit/DepositRecord.test.ts [hunk 1]
```diff
@@ -169,6 +152,59 @@ describe('=> DepositRecord', () => {
...
+
+      await expect(tx).revertedWith('Global deposit cap exceeded')
+    })
+
+    it('reverts if global caps already exceeded', async () => {
+      await depositRecord.connect(deployer).setGlobalNetDepositCap(TEST_AMOUNT_TWO)
+      await depositRecord.connect(user).recordDeposit(user.address, TEST_AMOUNT_TWO)
+      await depositRecord.connect(deployer).setGlobalNetDepositCap(TEST_AMOUNT_TWO.sub(1))
+
+      const tx = depositRecord.connect(user).recordDeposit(user.address, TEST_AMOUNT_ONE)
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/DepositRecord.test.ts:168]:
```suggestion
      const tx = depositRecord.connect(user).recordDeposit(user.address, 0)
```

File: apps/smart-contracts/core/test/unit/DepositRecord.test.ts [hunk 2]
```diff
@@ -169,6 +152,59 @@ describe('=> DepositRecord', () => {
...
+      await expect(tx).revertedWith('Global deposit cap exceeded')
+    })
+
+    it('reverts if uncapped user and global caps already exceeded', async () => {
+      bypassList.isIncluded.whenCalledWith(uncappedUser.address).returns(true)
+      await depositRecord.connect(deployer).setGlobalNetDepositCap(TEST_AMOUNT_TWO)
+      await depositRecord.connect(user).recordDeposit(uncappedUser.address, TEST_AMOUNT_TWO)
+      await depositRecord.connect(deployer).setGlobalNetDepositCap(TEST_AMOUNT_TWO.sub(1))
+
+      const tx = depositRecord.connect(user).recordDeposit(uncappedUser.address, TEST_AMOUNT_ONE)
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/DepositRecord.test.ts:179]:
```suggestion
      const tx = depositRecord.connect(user).recordDeposit(uncappedUser.address, 0)
```

File: apps/smart-contracts/core/test/unit/DepositRecord.test.ts [hunk 3]
```diff
@@ -169,6 +152,59 @@ describe('=> DepositRecord', () => {
...
+
+      await expect(tx).revertedWith('Global deposit cap exceeded')
+    })
+
+    it('reverts if user caps already exceeded', async () => {
+      await depositRecord.connect(deployer).setUserDepositCap(TEST_AMOUNT_TWO)
+      await depositRecord.connect(user).recordDeposit(user.address, TEST_AMOUNT_TWO)
+      await depositRecord.connect(deployer).setUserDepositCap(TEST_AMOUNT_TWO.sub(1))
+
+      const tx = depositRecord.connect(user).recordDeposit(user.address, TEST_AMOUNT_ONE)
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/DepositRecord.test.ts:189]:
```suggestion
      const tx = depositRecord.connect(user).recordDeposit(user.address, 0)
```

File: apps/smart-contracts/core/test/unit/DepositRecord.test.ts [hunk 4]
```diff
@@ -169,6 +152,59 @@ describe('=> DepositRecord', () => {
...
+    it('reverts if uncapped user changes to capped user and user caps already exceeded', async () => {
+      await depositRecord.connect(deployer).setUserDepositCap(TEST_AMOUNT_ONE)
+      bypassList.isIncluded.whenCalledWith(uncappedUser.address).returns(true)
+      await depositRecord.connect(user).recordDeposit(uncappedUser.address, TEST_AMOUNT_TWO)
+      expect(await depositRecord.getUserDepositCap()).lt(
+        await depositRecord.getUserDepositAmount(uncappedUser.address)
+      )
+      bypassList.isIncluded.whenCalledWith(uncappedUser.address).returns(false)
+
+      const tx = depositRecord.connect(user).recordDeposit(uncappedUser.address, TEST_AMOUNT_ONE)
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/DepositRecord.test.ts:203]:
```suggestion
      const tx = depositRecord.connect(user).recordDeposit(uncappedUser.address, 0)
```

File: apps/smart-contracts/core/test/unit/DepositRecord.test.ts [hunk 5]
```diff
@@ -169,6 +152,59 @@ describe('=> DepositRecord', () => {
       await expect(tx).revertedWith('Global deposit cap exceeded')
     })
 
+    it('reverts if amount > global cap', async () => {
+      await depositRecord.connect(deployer).setGlobalNetDepositCap(TEST_AMOUNT_TWO)
+
+      const tx = depositRecord.connect(user).recordDeposit(user.address, TEST_AMOUNT_TWO.add(1))
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/unit/DepositRecord.test.ts:158]: if there's only one test value, just use `TEST_AMOUNT_ONE` - same for tests below (it'll be the case after using 0 instead)

**Review by XavierEkkel — APPROVED** [review]
LGTM after nits

**Pull request merged**